substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -2473,6 +2513,844 @@ static boolean not(byte b) {
+                    //  2073               */
+                    //  2074  #if defined(__linux__) /* elided */
+                    //  2083  #endif
+                    //  2084              if (errno) {
+                    if (Errno.errno() > 0) {
+                        //  2085                  if (join) {
+                        if (join) {
+                            //  2086                      NET_ThrowCurrent(env, "setsockopt IP_ADD_MEMBERSHIP failed");
+                            throw new SocketException(PosixUtils.lastErrorString("setsockopt IP_ADD_MEMBERSHIP failed"));
+                            //  2087                  } else {
+                        } else {
+                            //  2088                      if (errno == ENOENT)
+                            if (Errno.errno() == Errno.ENOENT()) {
+                                //  2089                          JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException",
+                                //  2090                              "Not a member of the multicast group");
+                                throw new SocketException("Not a member of the multicast group");
+                                //  2091                      else
+                            } else {
+                                //  2092                          NET_ThrowCurrent(env, "setsockopt IP_DROP_MEMBERSHIP failed");
+                                throw new SocketException(PosixUtils.lastErrorString("setsockopt IP_DROP_MEMBERSHIP failed"));
+                                // no {} in original code if/else
+                            }
+                            //  2093                  }
+                        }
+                        //  2094                  return;
+                        // not reached
+                        //  2095              }
+                    }
+                    //  2096          }
+                }
+                //  2097
+                //  2098          /*
+                //  2099           * If we haven't switched to IPv6 socket option then we're done.
+                //  2100           */
+                //  2101          if (!ipv6_join_leave) {
+                if (! ipv6_join_leave) {
+                    //  2102              return;
+                    return;
+                    //  2103          }
+                }
+                //  2104      }
+            }
+            //  2105
+            //  2106
+            //  2107      /*
+            //  2108       * IPv6 join. If it's an IPv4 multicast group then we use an IPv4-mapped
+            //  2109       * address.
+            //  2110       */
+            //  2111      {
+            {
+                //  2112          struct ipv6_mreq mname6;
+                NetinetIn.ipv6_mreq mname6 = StackValue.get(NetinetIn.ipv6_mreq.class);
+                //  2113          jbyteArray ipaddress;
+                byte[] ipaddress;
```

Comment:
[The value of the local variable ipaddress is not used.](https://travis-ci.org/oracle/graal/jobs/548381854#L2745)